t/mojo/reactor_ev.t

```diff
@@ -8,307 +8,387 @@ plan skip_all => 'EV 4.32+ required for this test!'                  unless eval
-ok $last,   'timers were triggered in the right order';
-
-# Reset timer
-my $before = steady_time;
-my ($after, $again);
-$one = $reactor->timer(300 => sub { $after = steady_time });
-$two = $reactor->recurring(
-  300 => sub {
-    my $reactor = shift;
-    $reactor->remove($two) if ++$again > 3;
-  }
-);
-$reactor->timer(
-  0.025 => sub {
-    my $reactor = shift;
-    $reactor->again($one, 0.025);
-    $reactor->again($two, 0.025);
-  }
-);
-$reactor->start;
-ok $after, 'timer was triggered';
-ok(($after - $before) < 200, 'less than 200 seconds');
-is $again, 4, 'recurring timer triggered four times';
-
-# Restart inactive timer
-$id = $reactor->timer(0 => sub { });
-ok $reactor->remove($id), 'removed';
-eval { $reactor->again($id) };
-like $@, qr/Timer not active/, 'right error';
-
-# Change inactive I/O watcher
-ok !$reactor->remove($listen), 'not removed again';
-eval { $reactor->watch($listen, 1, 1) };
-like $@, qr!I/O watcher not active!, 'right error';
-
-# Error
-my $err;
-$reactor->unsubscribe('error')->on(
-  error => sub {
-    shift->stop;
-    $err = pop;
-  }
-);
-$reactor->timer(0 => sub { die "works!\n" });
-$reactor->start;
-like $err, qr/works!/, 'right error';
-
-# Reset events
-$reactor->on(error => sub { });
-ok $reactor->has_subscribers('error'), 'has subscribers';
-$reactor->reset;
-ok !$reactor->has_subscribers('error'), 'no subscribers';
-
-# Recursion
-$timer   = undef;
-$reactor = $reactor->new;
-$reactor->timer(0 => sub { ++$timer and shift->one_tick });
-$reactor->one_tick;
-is $timer, 1, 'timer was triggered once';
-
-# Detection
-is(Mojo::Reactor->detect, 'Mojo::Reactor::EV', 'right class');
-
-# Reactor in control
-is ref Mojo::IOLoop->singleton->reactor, 'Mojo::Reactor::EV', 'right object';
-ok !Mojo::IOLoop->is_running, 'loop is not running';
-my ($buffer, $server_err, $server_running, $client_err, $client_running);
-$id = Mojo::IOLoop->server(
-  {address => '127.0.0.1'} => sub {
-    my ($loop, $stream) = @_;
-    $stream->write('test' => sub { shift->write('321') });
-    $server_running = Mojo::IOLoop->is_running;
-    eval { Mojo::IOLoop->start };
-    $server_err = $@;
-  }
-);
-$port = Mojo::IOLoop->acceptor($id)->port;
-Mojo::IOLoop->client(
-  {port => $port} => sub {
-    my ($loop, $err, $stream) = @_;
-    $stream->on(
-      read => sub {
-        my ($stream, $chunk) = @_;
-        $buffer .= $chunk;
-        return unless $buffer eq 'test321';
-        Mojo::IOLoop->singleton->reactor->stop;
-      }
-    );
-    $client_running = Mojo::IOLoop->is_running;
-    eval { Mojo::IOLoop->start };
-    $client_err = $@;
-  }
-);
-Mojo::IOLoop->singleton->reactor->start;
-ok !Mojo::IOLoop->is_running, 'loop is not running';
-like $server_err, qr/^Mojo::IOLoop already running/, 'right error';
-like $client_err, qr/^Mojo::IOLoop already running/, 'right error';
-ok $server_running, 'loop is running';
-ok $client_running, 'loop is running';
+
+subtest 'With reactor' => sub {
```

Comment:
Does all the subtests need to be wrapped in another subtest if there's no global variables?

Reply:
Unneeded wrapper subtest is removed now.